http/http_response.hpp

```diff
@@ -206,8 +206,8 @@ struct Response
         }
         response.content_length(*pSize);
 
-        if (is1XXReturn || result() == status::no_content ||
-            result() == status::not_modified)
+        if ((*pSize > 0) && (is1XXReturn || result() == status::no_content ||
```

Comment:
Can you explain more in the commit why this is needed ?

Reply:
I this code is good. Due to  https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/304 "The response must not contain a body". I assume 1xx codes are the same.